doc/machine-auto.rst

```diff
@@ -0,0 +1,33 @@
+machine_dict: 
+    | type: ``dict``
+    | argument path: ``machine_dict``
+
+    batch_type: 
+        | type: ``str``
+        | argument path: ``machine_dict/batch_type``
+
+        The batch job system type. Option: Slurm, PBS, LSF, Shell, DpCloudServer
+
+    context_type: 
+        | type: ``str``
+        | argument path: ``machine_dict/context_type``
+
+        The connection used to remote machine. Option: LocalContext, LazyLocalContext, SSHContext， DpCloudServerContext
+
+    local_root: 
+        | type: ``str``
+        | argument path: ``machine_dict/local_root``
+
+        The dir where the tasks and relating files locate. Typically the project dir.
+
+    remote_root: 
+        | type: ``str``, optional
+        | argument path: ``machine_dict/remote_root``
+
+        The dir where the tasks are executed on the remote machine.
```

Comment:
consider adding something like the following: "Only needed when content_type is not lazy-local."